src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -68,82 +66,55 @@ internal class RenameEndpoint : IRenameEndpoint
+            var documentContext = await _documentContextFactory.CreateAsync(request.TextDocument.Uri, cancellationToken);
             var sourceText = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
-            if (!request.Position.TryGetAbsoluteIndex(sourceText, _logger, out var absoluteIndex))
-            {
-                return null;
-            }
-
+            var absoluteIndex = request.Position.GetRequiredAbsoluteIndex(sourceText, _logger);
             var projection = await _documentMappingService.GetProjectionAsync(documentContext, absoluteIndex, cancellationToken).ConfigureAwait(false);
 
-            // If we're in C# then there is no point checking for a component tag, because there won't be one
-            if (projection.LanguageKind != RazorLanguageKind.CSharp)
-            {
-                var razorEdits = await TryGetRazorComponentRenameEditsAsync(request, absoluteIndex, requestDocumentSnapshot, codeDocument, cancellationToken).ConfigureAwait(false);
-                if (razorEdits is not null)
-                {
-                    return razorEdits;
-                }
-            }
-
-            // If we're not doing single server rename, then we're done. C# and Html will be handled by the RenameHandler in the HtmlCSharp server.
-            if (!_languageServerFeatureOptions.SingleServerSupport)
-            {
-                return null;
-            }
-
-            // If the language is Razor then the downstream servers won't know how to handle it anyway
-            if (projection.LanguageKind == Protocol.RazorLanguageKind.Razor)
-            {
-                return null;
-            }
-
-            var delegatedParams = new DelegatedRenameParams(
-                documentContext.Identifier,
-                projection.Position,
-                projection.LanguageKind,
-                request.NewName);
-            var delegatedRequest = await _languageServer.SendRequestAsync(LanguageServerConstants.RazorRenameEndpointName, delegatedParams).ConfigureAwait(false);
-            var delegatedResponse = await delegatedRequest.Returning<WorkspaceEdit?>(cancellationToken).ConfigureAwait(false);
+            return new DelegatedRenameParams(
+                        documentContext.Identifier,
+                        projection.Position,
+                        projection.LanguageKind,
+                        request.NewName);
+        }
 
-            if (delegatedResponse is not null)
+        /// <inheritdoc/>
+        protected override async Task<WorkspaceEdit?> HandleDelegatedResponseAsync(WorkspaceEdit? response, DocumentContext documentContext, CancellationToken cancellationToken)
+        {
+            if (response is null)
             {
-                delegatedResponse = await _documentMappingService.RemapWorkspaceEditAsync(delegatedResponse, cancellationToken).ConfigureAwait(false);
+                throw new ArgumentNullException(nameof(response));
```

Comment:
Is this right? If we don't get a response back from the delegation, I'd expect we would just let it flow back up and return null

Reply:
This is where I wish I had a bit more control, or I'm missing something with generics. I want to have this override know that null will not be given as part of the contract. I couldn't figure out an easy way to do that

Reply:
Can't you just mark the parameter as not being nullable?

Reply:
ohhh i see, your generic param is `WorkspaceEdit?` not `WorkspaceEdit`. hmm.. yeah, okay.. maybe just a comment then "The base class will never call us in this case" or something.